### PR TITLE
Dependency management

### PR DESCRIPTION
**Pull request by alxbenz**
Changes to resourceLoader to use promises to resolve dependencies.

Please have a look and comment or improve upon the code.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "resourceLoader",
   "homepage": "https://github.com/virtualidentityag/conditional-resource-loader",
-  "version": "0.0.8",
+  "version": "0.1.0",
```
Comment (lhwparis) [inline bower.json]: there is a missmatch with the versions we have tagged in github. we are already on 1.0.1 there.

File: demo/demo.html [hunk 2]
```diff
@@ -29,12 +30,87 @@
 		</script>
 	</head>
 	<body>
-		<div class="demo1" data-init="(function($elm) { $elm.plugin(); })" data-resources="[{test: (function() {return true;}), paths: ['js/jquery.plugin.js', 'css/demo.css', 'css/demo1.css']}]">
+<!--
```
Comment (lhwparis) [inline demo/demo.html]: indentation seems to be broken. mix of tabs and spaces. can you place our default .editorconfig in this repo and reformat?

**Review by lhwparis — CHANGES_REQUESTED** [review]

File: resourceLoader.js [hunk 3]
```diff
@@ -8,124 +8,281 @@ function resourceLoader(options) {
...
-								if (resource.base.substring(0, 2) === '##') {
-									resource.base = options.baseMap[resource.base];
-								}
-								path = resource.base + path;
-							} else if (options && options.base) {
-								path = options.base + path;
-							}
-						}
+        if (test) {
+          $.each(resource.paths, function() {
```
Comment (lhwparis) [inline resourceLoader.js]: indentation seems to be broken (tabs and spaces) can you put our default editorconfig in this repo and reformat?

File: resourceLoader.js [hunk 4]
```diff
@@ -8,124 +8,281 @@ function resourceLoader(options) {
...
+	if (_loader.conditionsAllArray.length > 0) {
 		getResources();
 		loadResources();
 	} else {
 		$(window).trigger('resourcesReady');
 	}
 }
+
+// creates absolute path from any given string
+function absolutePath(urlString) {
```
Comment (smohadjer) [inline resourceLoader.js]: The logic for absolute path handling doesn't seem robust enough to me. If I run the demo page from http://localhost/conditional-resource-loader/demo/demo.html or from http://localhost:8080/demo/demo.html, in both cases all resources, including those using relative paths, fail to load due to wrongly normalized paths.

Comment (alxbenz) [reply resourceLoader.js]: I adjusted the path handling to consider the port. So http://localhost:8080/demo/demo.html should be working now. I couldn't reproduce your behaviour for http://localhost/conditional-resource-loader/demo/demo.html.
Do you have any idea to get the path handling more robust?

**Comment by janrembold** [comment]
Looks fine to me.

File: demo/demo-debug.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		<div class="demo4"
+			 data-init="(function($elm) { $elm.component2(); $elm.component1(); })"
+			 data-resources="
+			 [
+			 	{
+			 		paths: ['js/jquery.component2.js', 'css/demo4.css'],
+			 		dependsOn: ['js/jquery.component1.js']
+			 	}
+			 ]">
+			Resources are loaded in order of dependencies: component1.js -> component2.js -> demo4.css
```
Comment (timomayer) [inline demo/demo-debug.html]: @smohadjer as discussed in the ticket js/jquery.component1.js will never be loaded so the dependecies wont either.

**Review by timomayer — APPROVED** [review]

**Pull request merged**